cognite/neat/rules/importers/_spreadsheet2rules.py

```diff
@@ -195,6 +196,7 @@ def to_rules(
         if user_rules and reference_rules:
             rules = user_rules
             rules.reference = reference_rules
+            issue_list = DataModelingScenarioValidator(rules, issue_list).validate()
```

Comment:
Hmm, this I am a bit uncertain about. I am wondering if it is better to simply have the validator directly on the rules class, and maybe reinitialize to trigger it, something like this
```python
user_rules = user_rules.model_dump()
user_rules["reference"] = reference_rules.model_dump()
rules = rules_cls.model_validate(user_rules)
```

I am a bit skeptical about introducing an external validator concept to the rules. Then, that is something you have to remember to run.

Reply:
Even better https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.validate_assignment